ls1intum/Artemis
# `General`: Add direct navigation between exercise detail view and assessment dashboard

### Checklist
#### General


#### Client
#### Changes affecting Programming Exercises

### Motivation and Context
To simplify navigation between the assessment dashboard and the exercise details, users should be able to navigate from the exercise assessment dashboard to the exercise details page. The user should also be able to navigate from the exercise details page to the exercise assessment dashboard if manual assessment or complaints are active. This resolves #6905 

### Description
This PR added a new button on the exercise assessment dashboard in the exercise information table for navigation. 
The existing button on the exercise details page was updated to be active only if the assessment dashboard is active and the wording is updated to reflect the functionality of the button.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Exercise with manual assessment/ active complaints and 1 Exercise without manual assessment / active complaints

1. Log in to Artemis
2. Navigate to the Exercise Assessment Dashboard
3. Verify that the "Exercise Details Button" in the exercise information table navigates to the correct exercise details page and then verify that the "Exercise Assessment Dashboard" Button navigates correctly back to the exercise assessment dashboard 
4. Navigate to the Exercise List and choose an exercise without  manual assessment / active complaints
5. Verify that there is no button to navigate to the exercise assessment dashboard


### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage







### Screenshots


<img width="1673" alt="Screenshot 2023-08-04 at 17 52 50" src="https://github.com/ls1intum/Artemis/assets/71335194/cb540c0d-9acd-4bb3-aa8c-4374bd2cf8e5">
<img width="1117" alt="Screenshot 2023-08-07 at 22 45 14" src="https://github.com/ls1intum/Artemis/assets/71335194/03eadba2-2ba5-4583-8249-7420f97f7a77">
<img width="1131" alt="Screenshot 2023-08-07 at 22 37 38" src="https://github.com/ls1intum/Artemis/assets/71335194/fe86a4ea-8062-481d-a61d-91d0d8ac0822">
